h2/src/main/org/h2/value/ValueJson.java

```diff
@@ -0,0 +1,180 @@
+package org.h2.value;
+
+import java.io.IOException;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.Iterator;
+
+import org.h2.message.DbException;
+import org.h2.util.Utils;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+
+public class ValueJson extends Value {
+    
+    private JsonNode json;
+    private String string;
+    private int binarySize;
+    
+    private static final ObjectMapper mapper = new ObjectMapper();
+    
+    ValueJson(String str) throws IOException {
+        int memFirst = Utils.getMemoryUsed();
```

Comment:
Do not use this method in the main code, it's not reliable.

Reply:
It's not entirely clear how to evaluate various json objects.
[Here](https://github.com/lazarevnik/h2database/commit/cbd2a7859f9264cb5adc2bb15ffb8b9139f66c12) used string length, but it may work not always.